__tests__/app.test.js

```diff
@@ -16,51 +16,145 @@ beforeEach(() => {
+      return request(app)
+        .get("/api/articles/999")
+        .expect(404)
+        .then(({ body: { msg } }) => {
+          expect(msg).toBe(`We can't find that!`);
         });
-        test(`Each object in the  returned array should have keys of 'slug' and 'description'`, () => {
-            return request(app)
-                .get("/api/topics")
-                .expect(200)
-                .then(({ body }) => {
-                    body.forEach((topic) => {
-                        expect(topic).toHaveProperty("slug");
-                        expect(topic).toHaveProperty("description");
-                    });
-                });
+    });
+    test("Should return status 400 and error message if article_id is invalid type", () => {
+      return request(app)
+        .get("/api/articles/bananaPancakes")
+        .expect(400)
+        .then(({ body: { msg } }) => {
+          expect(msg).toBe(`That input is invalid`);
         });
-        test("Each object in the returned array should only have 2 keys", () => {
-            return request(app)
-                .get("/api/topics")
-                .expect(200)
-                .then(({ body }) => {
-                    body.forEach((topic) => {
-                        expect(Object.keys(topic).length).toBe(2);
-                    });
-                });
+    });
+  });
+  describe("GET /api/articles/:article_id/comments", () => {
+    test("Should return 200 status code", () => {
+      return request(app).get("/api/articles/1/comments")
+        .expect(200);
+    });
+    test("Should return an array of comment objects with desired keys", () => {
+      const expectedShape = {
+        comment_id: 10,
+        body: "git push origin master",
+        votes: 0,
+        author: "icellusedkars",
+        article_id: 3,
+        created_at: "2020-06-20T07:24:00.000Z",
+      };
+      return request(app)
+        .get("/api/articles/3/comments")
+        .then(({ body }) => {
+          expect(body[1]).toMatchObject(expectedShape)
+        })
+    });
+    test("Should serve comments in order of most recent first", () => {
+      return request(app)
+        .get("/api/articles/1/comments")
+        .then(({ body }) => {
+          expect(body[0].comment_id).toBe(5);
+          expect(body[10].comment_id).toBe(9);
         });
+    });
+    test("Should send code 400 if invalid article_id is entered", () => {
+      return request(app)
+        .get("/api/articles/burritoSupreme/comments")
+        .expect(400)
+        .then(({ body }) => {
+          expect(body.msg).toBe(`burritoSupreme is not a valid article_id`);
+        })
+    });
+    test("Should send code 404 if article_id has no comments", () => {
+      return request(app)
+        .get("/api/articles/8/comments")
+        .expect(404)
+        .then(({ body }) => {
+          expect(body.msg).toBe(`We couldn't find any comments on that...`);
```

Comment:
Please reconsider whether this scenario should be an error. Some articles may not have comments yet and we would subsequently expect the comments array to be empty, but this wouldn't be an error as this is behaviour we want to see.